### DISCUSSION TITLE
Anyone Noticed a Lag When Playing Song?

### DISCUSSION THREAD

**Discussion by Andrew-McGee** [ampache/ampache#3089] [General]
I am on the latest develop. and the past couple of weeks I've noticed a lag before a song plays. Even on my home networked (wired) it now takes a few seconds for a track to start playing.  Walking around using my phone it can sometimes take much longer.

Previously I never had this issue and nothing changed in my front end code that would cause this but it is still possible something other than Ampache could be causing this. (PHP or Apache update for example).

I thought I would mention it here rather than raise an issue and see if anyone else has noticed this.  I'm still investigating - will post here if I find anything.

**Comment by lachlan-00** [comment]
Possibly the change of getting similar tracks. I'll pull it out tomorrow morning.

**Comment by lachlan-00** [comment]
7fa13fc4a should stop that

**Comment by Andrew-McGee** [comment]
Yes this has resolved the lag issue and reverted back to previous performance.

The similar track recommendation is a great feature and something I wanted to incorporate in my front end. How could we implement it differently?

I'd love to see it in the API as a call that can happen asynch so it doesn't slow down playback.  Also, I am going to build my own buffering so I can implement gapless playback so that would be a work around.

**Reply by lachlan-00** [reply]
usually run:catalogUpdate -u should get the recommendations but it does do a lot.

I think extending the api to get that data as a pre type (artist/album/song) will help
